pkg/ottl/ottlfuncs/func_get_xml.go

```diff
@@ -0,0 +1,63 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs // import "github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/ottlfuncs"
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/antchfx/xmlquery"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+type GetXMLArguments[K any] struct {
+	Target ottl.StringGetter[K]
+	XPath  string
+}
+
+func NewGetXMLFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("GetXML", &GetXMLArguments[K]{}, createGetXMLFunction[K])
+}
+
+func createGetXMLFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*GetXMLArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("GetXML args must be of type *GetXMLAguments[K]")
+	}
+
+	if err := validateXPath(args.XPath); err != nil {
+		return nil, err
+	}
+
+	return getXML(args.Target, args.XPath), nil
+}
+
+// getXML returns a XML formatted string that is a result of removing all matching nodes from the target XML.
+// This currently supports removal of elements, attributes, text values, comments, and CharData.
```

Comment:
This line seems to contradict the line in the README that claims only selecting is supported.

Reply:
Good catch. Just a copy/paste error. Updated now.